src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Managed.cs

```diff
@@ -92,8 +93,13 @@ public void Abort()
                 handler.Credentials = options.Credentials;
                 handler.Proxy = options.Proxy;
                 handler.CookieContainer = options.Cookies;
+                handler.SslOptions.RemoteCertificateValidationCallback = options.RemoteCertificateValidationCallback;
                 if (options._clientCertificates?.Count > 0) // use field to avoid lazily initializing the collection
                 {
+                    if (handler.SslOptions.ClientCertificates == null)
+                    {
+                        handler.SslOptions.ClientCertificates = new X509CertificateCollection();
```

Comment:
Why not make this an `X509Certificate2Collection`  which is more useful.

Reply:
You mean why not change the type of SslClientAuthenticationOptions?

Reply:
>You mean why not change the type of SslClientAuthenticationOptions?

That would be nice too.  But doing a 'new X509Certificate2Collection()` is still good because it can still be assigned to the property which has type `X509CertificateCollection`

Reply:
What benefit does that provide? handler.SslOptions.ClientCertificates is typed as X509CertificateCollection, and options.ClientCertificates is typed as X509CertificateCollection.

Reply:
X509Certificate2Collection supports the Find() method, etc.  I don't understand why we are building new APIs with the older collection type.

@bartonjs can you comment as well?

Reply:
> X509Certificate2Collection supports the Find() method, etc.

But we're not using any of that.

> I don't understand why we are building new APIs with the older collection type.

Are you talking about this code constructing an X509CertificateCollection instead of an X509Certificate2Collection, or are you talking about the SslStream API exposing a property typed as X509CertificateCollection?  I can't speak to the latter.  For the former, X509Certificate2Collection derives from X509CertificateCollection, as far as I can tell overrides zero behaviors from the base, and the only purpose of those code is to ensure that a property typed as X509CertificateCollection is initialized to be non-null so we can add to it.  I can change it to construct an X509Certificate2Collection instead of an X509CertificateCollection, but as far as I can tell there's zero value in doing so.

Reply:
The example I use is that the collection in WinHttpHandler for example for ClientCertificates is an X509Certificate2Collection. It uses the Find() APIs to filter down certificates.

The previous recommendation from @bartonjs was to use X509Certificate2Collection and stop using X509CertificateCollection.  So, I'm surprised that we create newer APIs using the older collection types.

Reply:
I don't recall having opinions about the collection type.  I assume this used the (v1) collection because https://msdn.microsoft.com/en-us/library/system.net.security.sslstream.authenticateasclient(v=vs.110).aspx accepts one as input.

X509Certificate2 should definitely be used instead of X509Certificate when creating cert objects.

Using an X509Certificate2Collection is nicer, because it stops the older cert class from being utilized.

> X509Certificate2Collection supports the Find() method, etc.

Ugh, please don't call Find().  It's a GC flak cannon.  (It doesn't (and for compat: can't) return any of the input objects, it makes new ones.  And no one pays attention to them, so they go off to the GC, and then get Finalized because they hold native resources)